src/Eshopworld.Web/NotificationObservableHost.cs

```
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+
+namespace Eshopworld.Web
+{
+    public class NotificationObservableHost : IDisposable
+    {
+        //so the requirement here is hot broadcasting (Rx) entity - observable observer -
+        //and out of the box subjects fits the bill (most if not all of its code would exist in custom implementation)
+        //if performance becomes concern, custom implementation may be created
+        private readonly Subject<object> _observable = new Subject<object>();
+
+        /// <summary>
+        /// new event to be ingested
+        /// </summary>
+        /// <param name="event">event instance</param>
+        public void NewEvent(object @event)
+        {
+            if (@event != null)
+            {
+                _observable.OnNext(@event);
+            }
+        }
+
+        /// <summary>
+        /// shortcut for subscribing to everything
+        /// telemetry purposes etc.
+        /// </summary>
+        /// <param name="subscriber">observing delegate</param>
+        public IDisposable SubscribeToAll(Action<object> subscriber)
+        {
+            return Subscribe<object>(subscriber);
+        }
+
+        /// <summary>
+        /// subscribe to specific types (and indeed subtypes)
+        /// </summary>
+        /// <typeparam name="T">target type to subscribe</typeparam>
+        /// <param name="subscriber">observing delegate</param>
+        /// <returns>subscription <see cref="IDisposable"/></returns>
+        public IDisposable Subscribe<T>(Action<T> subscriber) where T : class
+        {
+            if (subscriber != null)
```

maybe it should throw an exception instead?